### PR TITLE
Bug fix. Displaying wrong error messages

### PR DESCRIPTION
**Pull request by guscarreon**
`parsePutObject()` returns an error, but those error messages were getting ignored because we were not exiting the  `put(...)` function immediately after the error check.

### CODE REVIEW

File: endpoints/put.go [hunk 1]
```diff
@@ -271,6 +271,7 @@ func (e *PutHandler) put(po *putObject, resp *putResponseObject, index int, wg *
 	toCache, err := parsePutObject(*po)
 	if err != nil {
 		resp.err = err
+		return
```
[the remark this answers is not in the document]
Comment (guscarreon) [reply endpoints/put.go:274]: done

File: endpoints/put_test.go [hunk 2]
```diff
@@ -1109,3 +1141,22 @@ func (b *deadlineExceedingBackend) Put(ctx context.Context, key string, value st
...
+}
+
+func (b *countingCallsBackend) Put(ctx context.Context, key string, value string, ttlSeconds int) error {
+	b.PutCalls++
+	return nil
+}
+
+func newCountingCallsBackend() *countingCallsBackend {
+	return &countingCallsBackend{}
+}
```
Comment (SyntaxNode) [inline endpoints/put_test.go:1162]: This call counting is prone to race conditions due to the parallel processing in the code. This is also a case better handled by using testify's mock framework with call count expectations.

Switching to this fake backend will always return empty strings, but the memory backend that was used before I think would return real data. Is that important for these tests? Were we validating the response of these methods?

File: endpoints/put_test.go [hunk 3]
```diff
@@ -989,30 +1034,17 @@ func doMockGet(t *testing.T, router *httprouter.Router, id string) *httptest.Res
...
 	}
-
 	router.ServeHTTP(rr, request)
-	uuid := ""
-	if rr.Code == http.StatusOK {
-		uuid = parseMockUUID(t, rr.Body.String())
-	}
-	return uuid, rr
+
+	return rr
```
Comment (SyntaxNode) [inline endpoints/put_test.go:1047]: Why did you remove the uuids and response parsing? That seems like worthwhile information. We were not using it before?

File: endpoints/put_test.go [hunk 4]
```diff
@@ -607,7 +648,7 @@ func TestBadPayloadSizePutError(t *testing.T) {
 
 func TestInternalPutClientError(t *testing.T) {
 	// Valid request
-	reqBody := "{\"puts\":[{\"type\":\"xml\",\"value\":\"text longer than size limit\"}]}"
+	reqBody := "{\"puts\":[{\"type\":\"xml\",\"value\":\"some data\"}]}"
```
Comment (SyntaxNode) [inline endpoints/put_test.go:651]: Is it no longer important (or was it never important?) for the value data to be longer than the size limit?

File: endpoints/put_test.go (hunk 3 above)
Comment (guscarreon) [reply endpoints/put_test.go:1012]: The first parameter was being ignored almost everywhere. In the one test that made use of it I simply extracted the UUID from the recorder body in the test itself.

File: endpoints/put_test.go (hunk 4 above)
Comment (guscarreon) [reply endpoints/put_test.go:617]: It was never important. `TestInternalPutClientError` asserts we handle an errors coming from the backend side (maybe even the storage service itself)

File: endpoints/put_test.go (hunk 2 above)
Comment (guscarreon) [reply endpoints/put_test.go:1122]: Resolved in a meeting, @SyntaxNode commited the changes that fix this race condition and removes this part of the code no longer necessary

File: endpoints/put_test.go [hunk 5]
```diff
@@ -593,10 +598,12 @@ func TestBadPayloadSizePutError(t *testing.T) {
 	m := metricstest.CreateMockMetrics()
 	router.POST("/cache", NewPutHandler(backend, m, 10, true))
 
-	_, httpTestRecorder := doMockPut(t, router, reqBody)
+	putResponse := doPut(t, router, reqBody)
+
+	// Assert expected response
+	assert.Equal(t, http.StatusBadRequest, putResponse.Code, "doMockPut should have failed when trying to store elements in sizeCappedBackend")
```
Comment (bsardo) [inline endpoints/put_test.go:604]: Nitpick: the string at the end refers to `doMockPut` but it should now be `doPut`

**Review by SyntaxNode — DISMISSED** [review]

File: endpoints/put_test.go [hunk 6]
```diff
@@ -482,22 +482,27 @@ func TestRequestReadError(t *testing.T) {
...
+
 	router := httprouter.New()
 	m := metricstest.CreateMockMetrics()
 	router.POST("/cache", NewPutHandler(backend, m, len(putElements)-1, true))
 
-	_, httpTestRecorder := doMockPut(t, router, reqBody)
-	assert.Equalf(t, http.StatusBadRequest, httpTestRecorder.Code, "doMockPut should have failed when trying to store %d elements because capacity is %d ", len(putElements), len(putElements)-1)
+	putResponse := doPut(t, router, reqBody)
+
+	assert.Equalf(t, http.StatusBadRequest, putResponse.Code, "doMockPut should have failed when trying to store %d elements because capacity is %d ", len(putElements), len(putElements)-1)
```
Comment (bsardo) [inline endpoints/put_test.go:501]: Nitpick: the string at the end refers to `doMockPut` but it should now be `doPut`

**Review by bsardo — COMMENTED** [review]
Just two nitpicks, otherwise looks good.

File: endpoints/put_test.go (hunk 6 above)
Comment (guscarreon) [reply endpoints/put_test.go:501]: Corrected

**Review by bsardo — APPROVED** [review]
LGTM
